### ISSUE TITLE
CLI shutdown throws an exeption on python <3.7 

### ISSUE THREAD

**Issue by gyorsok** [beveradb/pysonofflan#13]
* pySonoffLAN version: 0.2.1
* Python version: 3.6.7
* Operating System: ubuntu 18.06

### Description

CLI command `pysonofflan -h 10.4.0.12 state` throws an exeption, that
asyncio.all_tasks() is not a valid attribute/method.
According to [https://docs.python.org/3/library/asyncio-task.html#asyncio.all_tasks](https://docs.python.org/3/library/asyncio-task.html#asyncio.all_tasks), it has been added in python 3.7.
tox.ini suggests, that 3.5, 3.6, 3.7 should all be supported.

Changing it to `asyncio.Task.all_tasks` seems to fix it.
 
### What I Did

```
pysonofflan --host 10.4.0.12 state
2019-02-09 21:53:20,168 - info: Initialising SonoffSwitch with host 10.4.0.12
2019-02-09 21:53:20,331 - info: == Device: XXX (10.4.0.12) ==
2019-02-09 21:53:20,332 - info: State: ON
Traceback (most recent call last):
  File "/usr/local/bin/pysonofflan", line 11, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/click/decorators.py", line 64, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/pysonofflan/cli.py", line 154, in state
    logger=logger
  File "/usr/local/lib/python3.6/dist-packages/pysonofflan/sonoffswitch.py", line 58, in __init__
    context=context
  File "/usr/local/lib/python3.6/dist-packages/pysonofflan/sonoffdevice.py", line 68, in __init__
    self.loop.run_until_complete(self.setup_connection())
  File "/usr/lib/python3.6/asyncio/base_events.py", line 473, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.6/dist-packages/pysonofflan/sonoffdevice.py", line 86, in setup_connection
    await self.client.receive_message_loop()
  File "/usr/local/lib/python3.6/dist-packages/pysonofflan/client.py", line 131, in receive_message_loop
    await self.event_handler(message)
  File "/usr/local/lib/python3.6/dist-packages/pysonofflan/sonoffdevice.py", line 160, in handle_message
    await self.callback_after_update(self)
  File "/usr/local/lib/python3.6/dist-packages/pysonofflan/sonoffswitch.py", line 170, in pre_callback_after_update
    await self.parent_callback_after_update(self)
  File "/usr/local/lib/python3.6/dist-packages/pysonofflan/cli.py", line 148, in state_callback
    device.shutdown_event_loop()
  File "/usr/local/lib/python3.6/dist-packages/pysonofflan/sonoffdevice.py", line 185, in shutdown_event_loop
    *asyncio.all_tasks(loop=self.loop),
AttributeError: module 'asyncio' has no attribute 'all_tasks'
Task was destroyed but it is pending!
task: <Task pending coro=<SonoffDevice.send_updated_params_loop() running at /usr/local/lib/python3.6/dist-packages/pysonofflan/sonoffdevice.py:113> wait_for=<Future pending cb=[<TaskWakeupMethWrapper object at 0x7efe5ba25618>()]>>
Exception ignored in: <coroutine object SonoffDevice.send_updated_params_loop at 0x7efe5ba0ad00>
RuntimeError: coroutine ignored GeneratorExit
```

**Comment by beveradb** [comment]
Thanks for highlighting! I've got to admit, I'm not experienced with `asyncio` and I didn't do extensive testing on Python 3.6, so I can see how this slipped through.

I've spent the last week working on a hobby project to learn how to write unit tests properly (and structure my code better to _support_ proper unit testing), so over the next few days I'll be refactoring pysonofflan substantially and writing tons of unit tests to give me and others more confidence in the implementation.
After that, anything like this which doesn't work on one of the Python versions would be picked up by `tox` as part of the test run before release.

TL;DR: this will be fixed soon, as part of a major effort to refactor and write tests - hopefully within the next week.

**Comment by beveradb** [comment]
Pretty sure this will be fixed by the great work mattsaxon has been doing in https://github.com/beveradb/pysonofflan/pull/48

**Comment by mattsaxon** [comment]
I haven't tested on 3.6, but the all_tasks method is no longer used in the PR associated to #48, so I agree this is likely fixed when that PR is merged.

**Comment by beveradb** [comment]
I think this should be resolved now with the release of version `0.3.0` - feel free to re-open if not!
